EEEEEEEEE
======================================================================
ERROR: test_list_int (__main__.Test01_weather_read_data_errorhandle)
*** Test01 **** FUNCTION CALL: weather.read_data(filename='test74653.dat') *** EXPECT: {} ***
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/parallels/Desktop/CPSC223P_CLASS/assignment_4-kiethoang1411/test.py", line 18, in test_list_int
    r = weather.read_data(filename='test74653.dat')
AttributeError: module 'weather' has no attribute 'read_data'

======================================================================
ERROR: test_list_int (__main__.Test02_weather_write_data)
*** Test02 **** FUNCTION CALL: weather.write_data(data={"20210203075501": {"t": 55, "h": 87, "r": 0}}, filename='test74653.dat') THEN weather.read_data(filename='test74653.dat') *** EXPECT: {"20210203075501": {"t": 55, "h": 87, "r": 0}} ***
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/parallels/Desktop/CPSC223P_CLASS/assignment_4-kiethoang1411/test.py", line 31, in test_list_int
    weather.write_data(data={"20210203075501": {"t": 55, "h": 87, "r": 0}}, filename='test74653.dat')
AttributeError: module 'weather' has no attribute 'write_data'

======================================================================
ERROR: test_list_int (__main__.Test03_max_temperature)
*** Test03 **** FUNCTION CALL: weather.max_temperature(data={"20210203075501": {"t": 55, "h": 87, "r": 0}, "20210203090602": {"t": 63, "h": 84, "r": 0}, "20210203102903": {"t": 71, "h": 79, "r": 0}, "20210203125504": {"t": 72, "h": 69, "r": 0}, "20210203183905": {"t": 59, "h": 75, "r": 0}, "20210205044406": {"t": 57, "h": 68, "r": 0.01}, "20210205083307": {"t": 65, "h": 63, "r": 0.05}, "20210205122208": {"t": 73, "h": 56, "r": 0.11}, "20210205161109": {"t": 74, "h": 60, "r": 0.19}}, date='20210205') *** EXPECT: 74 ***
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/parallels/Desktop/CPSC223P_CLASS/assignment_4-kiethoang1411/test.py", line 43, in test_list_int
    r = weather.max_temperature(data={"20210203075501": {"t": 55, "h": 87, "r": 0}, "20210203090602": {"t": 63, "h": 84, "r": 0}, "20210203102903": {"t": 71, "h": 79, "r": 0}, "20210203125504": {"t": 72, "h": 69, "r": 0}, "20210203183905": {"t": 59, "h": 75, "r": 0}, "20210205044406": {"t": 57, "h": 68, "r": 0.01}, "20210205083307": {"t": 65, "h": 63, "r": 0.05}, "20210205122208": {"t": 73, "h": 56, "r": 0.11}, "20210205161109": {"t": 74, "h": 60, "r": 0.19}}, date='20210205')
AttributeError: module 'weather' has no attribute 'max_temperature'

======================================================================
ERROR: test_list_int (__main__.Test04_min_temperature)
*** Test04 **** FUNCTION CALL: weather.min_temperature(data={"20210203075501": {"t": 55, "h": 87, "r": 0}, "20210203090602": {"t": 63, "h": 84, "r": 0}, "20210203102903": {"t": 71, "h": 79, "r": 0}, "20210203125504": {"t": 72, "h": 69, "r": 0}, "20210203183905": {"t": 59, "h": 75, "r": 0}, "20210205044406": {"t": 57, "h": 68, "r": 0.01}, "20210205083307": {"t": 65, "h": 63, "r": 0.05}, "20210205122208": {"t": 73, "h": 56, "r": 0.11}, "20210205161109": {"t": 74, "h": 60, "r": 0.19}}, date='20210205') *** EXPECT: 57 ***
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/parallels/Desktop/CPSC223P_CLASS/assignment_4-kiethoang1411/test.py", line 53, in test_list_int
    r = weather.min_temperature(data={"20210203075501": {"t": 55, "h": 87, "r": 0}, "20210203090602": {"t": 63, "h": 84, "r": 0}, "20210203102903": {"t": 71, "h": 79, "r": 0}, "20210203125504": {"t": 72, "h": 69, "r": 0}, "20210203183905": {"t": 59, "h": 75, "r": 0}, "20210205044406": {"t": 57, "h": 68, "r": 0.01}, "20210205083307": {"t": 65, "h": 63, "r": 0.05}, "20210205122208": {"t": 73, "h": 56, "r": 0.11}, "20210205161109": {"t": 74, "h": 60, "r": 0.19}}, date='20210205')
AttributeError: module 'weather' has no attribute 'min_temperature'

======================================================================
ERROR: test_list_int (__main__.Test05_max_humidity)
*** Test05 **** FUNCTION CALL: weather.max_humidity(data={"20210203075501": {"t": 55, "h": 87, "r": 0}, "20210203090602": {"t": 63, "h": 84, "r": 0}, "20210203102903": {"t": 71, "h": 79, "r": 0}, "20210203125504": {"t": 72, "h": 69, "r": 0}, "20210203183905": {"t": 59, "h": 75, "r": 0}, "20210205044406": {"t": 57, "h": 68, "r": 0.01}, "20210205083307": {"t": 65, "h": 63, "r": 0.05}, "20210205122208": {"t": 73, "h": 56, "r": 0.11}, "20210205161109": {"t": 74, "h": 60, "r": 0.19}}, date='20210205') *** EXPECT: 68 ***
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/parallels/Desktop/CPSC223P_CLASS/assignment_4-kiethoang1411/test.py", line 63, in test_list_int
    r = weather.max_humidity(data={"20210203075501": {"t": 55, "h": 87, "r": 0}, "20210203090602": {"t": 63, "h": 84, "r": 0}, "20210203102903": {"t": 71, "h": 79, "r": 0}, "20210203125504": {"t": 72, "h": 69, "r": 0}, "20210203183905": {"t": 59, "h": 75, "r": 0}, "20210205044406": {"t": 57, "h": 68, "r": 0.01}, "20210205083307": {"t": 65, "h": 63, "r": 0.05}, "20210205122208": {"t": 73, "h": 56, "r": 0.11}, "20210205161109": {"t": 74, "h": 60, "r": 0.19}}, date='20210205')
AttributeError: module 'weather' has no attribute 'max_humidity'

======================================================================
ERROR: test_list_int (__main__.Test06_min_humidity)
*** Test06 **** FUNCTION CALL: weather.min_humidity(data={"20210203075501": {"t": 55, "h": 87, "r": 0}, "20210203090602": {"t": 63, "h": 84, "r": 0}, "20210203102903": {"t": 71, "h": 79, "r": 0}, "20210203125504": {"t": 72, "h": 69, "r": 0}, "20210203183905": {"t": 59, "h": 75, "r": 0}, "20210205044406": {"t": 57, "h": 68, "r": 0.01}, "20210205083307": {"t": 65, "h": 63, "r": 0.05}, "20210205122208": {"t": 73, "h": 56, "r": 0.11}, "20210205161109": {"t": 74, "h": 60, "r": 0.19}}, date='20210205') *** EXPECT: 56 ***
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/parallels/Desktop/CPSC223P_CLASS/assignment_4-kiethoang1411/test.py", line 73, in test_list_int
    r = weather.min_humidity(data={"20210203075501": {"t": 55, "h": 87, "r": 0}, "20210203090602": {"t": 63, "h": 84, "r": 0}, "20210203102903": {"t": 71, "h": 79, "r": 0}, "20210203125504": {"t": 72, "h": 69, "r": 0}, "20210203183905": {"t": 59, "h": 75, "r": 0}, "20210205044406": {"t": 57, "h": 68, "r": 0.01}, "20210205083307": {"t": 65, "h": 63, "r": 0.05}, "20210205122208": {"t": 73, "h": 56, "r": 0.11}, "20210205161109": {"t": 74, "h": 60, "r": 0.19}}, date='20210205')
AttributeError: module 'weather' has no attribute 'min_humidity'

======================================================================
ERROR: test_list_int (__main__.Test07_tot_rain)
*** Test07 **** FUNCTION CALL: weather.tot_rain(data={"20210203075501": {"t": 55, "h": 87, "r": 0}, "20210203090602": {"t": 63, "h": 84, "r": 0}, "20210203102903": {"t": 71, "h": 79, "r": 0}, "20210203125504": {"t": 72, "h": 69, "r": 0}, "20210203183905": {"t": 59, "h": 75, "r": 0}, "20210205044406": {"t": 57, "h": 68, "r": 0.01}, "20210205083307": {"t": 65, "h": 63, "r": 0.05}, "20210205122208": {"t": 73, "h": 56, "r": 0.11}, "20210205161109": {"t": 74, "h": 60, "r": 0.19}}, date='20210205') *** EXPECT: 0.36 ***
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/parallels/Desktop/CPSC223P_CLASS/assignment_4-kiethoang1411/test.py", line 83, in test_list_int
    r = weather.tot_rain(data={"20210203075501": {"t": 55, "h": 87, "r": 0}, "20210203090602": {"t": 63, "h": 84, "r": 0}, "20210203102903": {"t": 71, "h": 79, "r": 0}, "20210203125504": {"t": 72, "h": 69, "r": 0}, "20210203183905": {"t": 59, "h": 75, "r": 0}, "20210205044406": {"t": 57, "h": 68, "r": 0.01}, "20210205083307": {"t": 65, "h": 63, "r": 0.05}, "20210205122208": {"t": 73, "h": 56, "r": 0.11}, "20210205161109": {"t": 74, "h": 60, "r": 0.19}}, date='20210205')
AttributeError: module 'weather' has no attribute 'tot_rain'

======================================================================
ERROR: test_list_int (__main__.Test08_report_daily)
*** Test08 **** FUNCTION CALL: weather.report_daily(data={"20210203075501": {"t": 55, "h": 87, "r": 0}, "20210203090602": {"t": 63, "h": 84, "r": 0}, "20210203102903": {"t": 71, "h": 79, "r": 0}, "20210203125504": {"t": 72, "h": 69, "r": 0}, "20210203183905": {"t": 59, "h": 75, "r": 0}, "20210205044406": {"t": 57, "h": 68, "r": 0.01}, "20210205083307": {"t": 65, "h": 63, "r": 0.05}, "20210205122208": {"t": 73, "h": 56, "r": 0.11}, "20210205161109": {"t": 74, "h": 60, "r": 0.19}}, date='20210205') *** EXPECT: <look in test.py Test08 variable = report> ***
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/parallels/Desktop/CPSC223P_CLASS/assignment_4-kiethoang1411/test.py", line 93, in test_list_int
    r = weather.report_daily(data={"20210203075501": {"t": 55, "h": 87, "r": 0}, "20210203090602": {"t": 63, "h": 84, "r": 0}, "20210203102903": {"t": 71, "h": 79, "r": 0}, "20210203125504": {"t": 72, "h": 69, "r": 0}, "20210203183905": {"t": 59, "h": 75, "r": 0}, "20210205044406": {"t": 57, "h": 68, "r": 0.01}, "20210205083307": {"t": 65, "h": 63, "r": 0.05}, "20210205122208": {"t": 73, "h": 56, "r": 0.11}, "20210205161109": {"t": 74, "h": 60, "r": 0.19}}, date='20210205')
AttributeError: module 'weather' has no attribute 'report_daily'

======================================================================
ERROR: test_list_int (__main__.Test09_report_historical)
*** Test09 **** FUNCTION CALL: weather.report_historical(data={"20210203075501": {"t": 55, "h": 87, "r": 0}, "20210203090602": {"t": 63, "h": 84, "r": 0}, "20210203102903": {"t": 71, "h": 79, "r": 0}, "20210203125504": {"t": 72, "h": 69, "r": 0}, "20210203183905": {"t": 59, "h": 75, "r": 0}, "20210205044406": {"t": 57, "h": 68, "r": 0.01}, "20210205083307": {"t": 65, "h": 63, "r": 0.05}, "20210205122208": {"t": 73, "h": 56, "r": 0.11}, "20210205161109": {"t": 74, "h": 60, "r": 0.19}}) *** EXPECT: <look in test.py Test09 variable = report> ***
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/parallels/Desktop/CPSC223P_CLASS/assignment_4-kiethoang1411/test.py", line 110, in test_list_int
    r = weather.report_historical(data={"20210203075501": {"t": 55, "h": 87, "r": 0}, "20210203090602": {"t": 63, "h": 84, "r": 0}, "20210203102903": {"t": 71, "h": 79, "r": 0}, "20210203125504": {"t": 72, "h": 69, "r": 0}, "20210203183905": {"t": 59, "h": 75, "r": 0}, "20210205044406": {"t": 57, "h": 68, "r": 0.01}, "20210205083307": {"t": 65, "h": 63, "r": 0.05}, "20210205122208": {"t": 73, "h": 56, "r": 0.11}, "20210205161109": {"t": 74, "h": 60, "r": 0.19}})
AttributeError: module 'weather' has no attribute 'report_historical'

----------------------------------------------------------------------
Ran 9 tests in 0.001s

FAILED (errors=9)
